### PR TITLE
[SALEOR-7179] Send SALEOR_DOMAIN alongside SALEOR_AUTH_TOKEN to environment variables

### PR DESCRIPTION
**Pull request by magul**

### CODE REVIEW

File: lib/middlewares.ts [hunk 1]
```diff
@@ -1,83 +1,92 @@
...
-  const saleorDomain = request.headers[Constants.SALEOR_DOMAIN_HEADER];
-  if (!saleorDomain) {
-    throw new MiddlewareError("Missing saleor domain token.", 400);
-  }
-  return saleorDomain;
-};
+    if (SALEOR_DOMAIN === undefined) {
+      return Response.InternalServerError({
+        success: false,
+        message: "Missing Saleor domain environment variable.",
```
Comment (Cloud11PL) [inline lib/middlewares.ts:27]: For the sake of clarity maybe let's specify which one?

File: pages/configuration.tsx [hunk 2]
```diff
@@ -24,41 +28,40 @@ const Configuration: NextPage = () => {
...
   useEffect(() => {
-    appState?.domain && appState?.token && fetch(
-        "/api/configuration",
-        { headers: [
+    appState?.domain &&
+      appState?.token &&
+      fetch("/api/configuration", {
+        headers: [
           [SALEOR_DOMAIN_HEADER, appState.domain],
           ["authorization-bearer", appState.token!],
```
Comment (Cloud11PL) [inline pages/configuration.tsx:40]:
```suggestion
          ["authorization-bearer", appState.token],
```

File: pages/orders.tsx [hunk 3]
```diff
@@ -126,65 +123,71 @@ const Orders: NextPage = () => {
...
             </TableHead>
             <TableBody>
-
-              {fetching && <TableRow>
-                <TableCell colSpan={6}>Loading...</TableCell>
-              </TableRow>}
-
+              {fetching && (
+                <TableRow>
+                  <TableCell colSpan={6}>Loading...</TableCell>
```
[the remark this answers is not in the document]
Comment (magul) [reply pages/orders.tsx:128]: From where do you want this `<Skeleton />` to be imported? I have found such component in dashboard. Should we copy it to template? Or move to Macaw-UI?

File: pages/orders.tsx [hunk 4]
```diff
@@ -76,20 +76,19 @@ const Orders: NextPage = () => {
...
-        const number_of_orders: string | null = json.data?.number_of_orders;
-        setNumberOfOrders(parseInt(number_of_orders || "100"));
-      });
+        ],
+      })
+        .then((res) => res.json())
+        .then((json) => {
+          const number_of_orders: string | null = json.data?.number_of_orders;
+          setNumberOfOrders(parseInt(number_of_orders || "100"));
+        });
```
[the remark this answers is not in the document]
Comment (magul) [reply pages/orders.tsx:91]: We cannot make it that way, as `json()` is async function (so either chain of promise resolutions - the way it is currently, or adding `await` in front of `res.json()`.

File: lib/environment.ts [hunk 5]
```diff
@@ -1,37 +1,60 @@
-import fs from "fs";
+import { promises as fsPromises } from "fs";
```
Comment (zaiste) [inline lib/environment.ts:1]: @magul you can use `fs-extra` as it defaults to promises and comes with some additional, handy functions.

**Review by zaiste — APPROVED** [review]

**Pull request merged**